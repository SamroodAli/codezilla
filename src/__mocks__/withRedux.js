import reduxThunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { StaticRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import PropTypes from 'prop-types';
import data from '../../public/data/courses.json';

const initialState = {
  videos: [data],
  filters: {
    React: true,
    Angular: true,
    Vue: true,
    'Functional Programming': true,
    'Object Oriented Programming': true,
    Microservices: true,
    'JAM Stack': true,
  },
  theme: 'light',
  currentVideo: null,
};

const mockStore = configureStore([reduxThunk]);
export const store = mockStore(initialState);

const WithRedux = ({ children }) => (
  <Provider store={store}><StaticRouter>{children}</StaticRouter></Provider>
);

WithRedux.propTypes = {
  children: PropTypes.node.isRequired,
};

export default WithRedux;
