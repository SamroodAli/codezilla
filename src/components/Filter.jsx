import useActions from '../hooks/useActions';

const Filter = () => {
  const { setFilter } = useActions();
  return (
    <form>
      <input type="radio" id="css" onClick={() => setFilter('React')} />
      <input type="radio" id="css" onClick={() => setFilter('Angular')} />
      <input type="radio" id="css" onClick={() => setFilter('Vue')} />
    </form>
  );
};

export default Filter;
