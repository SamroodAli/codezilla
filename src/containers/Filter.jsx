import { useSelector } from 'react-redux';
import useActions from '../hooks/useActions';
import Filter from '../components/Filter';

const FilterContainer = () => {
  const { setFilter } = useActions();
  const filterState = useSelector(({ filters }) => filters);
  return <Filter filterState={filterState} setFilter={setFilter} />;
};

export default FilterContainer;
