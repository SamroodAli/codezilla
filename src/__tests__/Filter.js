import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import withRedux from 'withRedux';
import Filter from '../components/Filter';

describe('Testing Filter with redux', () => {
  it('Filter is defined', async () => {
    const courses = render(withRedux(<Filter />));
    expect(courses).toBeDefined();
  });

  it('Filter snapshot testing', () => {
    const tree = renderer.create(withRedux(<Filter />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
