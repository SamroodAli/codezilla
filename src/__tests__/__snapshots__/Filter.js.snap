// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Filter with redux Filter snapshot testing 1`] = `
Array [
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        React
      </div>
    </label>
  </label>,
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        Vue
      </div>
    </label>
  </label>,
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        Angular
      </div>
    </label>
  </label>,
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        Functional Programming
      </div>
    </label>
  </label>,
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        Microservices
      </div>
    </label>
  </label>,
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        JAM Stack
      </div>
    </label>
  </label>,
  <label
    htmlFor="React"
  >
    <label
      className=""
      data-baseweb="checkbox"
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      title={null}
    >
      <span
        className=""
      />
      <input
        aria-checked={true}
        aria-invalid={null}
        aria-required={null}
        checked={true}
        className=""
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <div
        className=""
      >
        Object Oriented Programming
      </div>
    </label>
  </label>,
]
`;
